// scrappers/siscop.js
var siscop = {
  loginIndex: 'cpf_senha.asp',
  host: 'https://siscop.portalcorporativo.serpro',
  STRING_EXIT: 'Sair com Seguran',
  STRING_MAT_ERROR: 'ERRO A90002 -',
  STRING_APPD_ERROR: 'ERRO A38001',
  STRING_INCORRECT: 'Procedimento incorreto',
  STRING_SERVER_ERROR: 'Server Error',
  login: function(login, password, mat) {

  },
  scrap: function(req, res, request, between, iconv) {
    const data = {
      viewstate: req.body.viewstate,
      tx_cpf: req.body.cpf,
      tx_senha: req.body.pass,
      opt_reload_captcha: 0,
      captcha: req.body.captcha
    };
    request.post({
      headers: {
        'content-type': 'application/x-www-form-urlencoded; charset=utf-8',
      },
      url: this.host + '/' + this.loginIndex,
      strictSSL: false,
      followAllRedirects: true,
      jar: true,
      encoding: null,
      rejectUnauthorized: false,
      form: data
    }, function(error, response, body) {
      body = iconv.convert(body).toString('utf-8');
      console.log(body);
      if (body.indexOf(siscop.STRING_EXIT) > 0) {
        //Login OK
        res.json({
          msg: 'OK'
        });
      } else {
        //LOGIN FAIL
        if (body.indexOf("<div class=\"erros\"") > 0) {
          var err = between(body,
            "<div class=\"erros\" role=\"alert\" aria-live=\"assertive\">\n    \t<h3>Aviso</h3>\n        <ul>\n           <li>",
            "</li>");
        } else {
          err = '';
        }
        res.json({
          error: 'FAIL',
          msg: err
            //result: body
        });
      }
      res.send();
    });
  }
};
module.exports = siscop;
